app/org/sagebionetworks/bridge/dynamodb/DynamoStudyDao.java

```diff
@@ -59,14 +60,39 @@ public Study getStudy(String identifier) {
         if (study == null) {
             throw new EntityNotFoundException(Study.class, "Study '"+identifier+"' not found.");
         }
+        if (!study.isActive()) {
+            throw new EntityNotFoundException(Study.class, "Study '"+identifier+"' is de-activated.");
+        }
+        return study;
+    }
+
+    /**
+     * Only used for dao and testing to get a study no matter if it is deactivated or not
+     * @param identifier
+     * @return
+     */
+    Study getStudyOnlyForDao(String identifier) {
+        checkArgument(isNotBlank(identifier), Validate.CANNOT_BE_BLANK, "identifier");
+
+        DynamoStudy study = new DynamoStudy();
+        study.setIdentifier(identifier);
+        study = mapper.load(study);
+        if (study == null) {
+            throw new EntityNotFoundException(Study.class, "Study '"+identifier+"' not found.");
+        }
+
         return study;
```

Comment:
As in getStudies() below, you could add an includeDeleted boolean flag to the existing getStudy() call, and locally use that method with true rather than false. This is less important since I don't believe you can get a study until you re-enable it, outside of the DAO itself.

(As discussed, the service can provide overloaded methods that always set this to false, so everything continues to compile.)

For the use case of an admin restoring a deactivated study, the admin needs to be able to retrieve all studies, including deactivated studies, and should be able to update a deactivated study with the "active" flag set to false (I don't think this needs to be exposed as a separate API endpoint, but note that setting the active flag should be protected by the "isAdminUpdate" flag in the StudyService). Then getStudy() and everything else will work again.